### PR TITLE
Implement resume for capturing traces

### PR DESCRIPTION
**Pull request by kralka**
Implements a helper iterator for (key, text) pairs which is able to
auto-save progress.

### CODE REVIEW

File: scaaml/io/resume_kti.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+  Progress file is a text file with a single integer -- the number of traces
+  captured so far. The number in it is a multiple of the shard length. This file
+  is automatically updated after shard length iterations over ResumeKTI.
+"""
+
+import numpy as np
+
+
+def create_resume_kti(keys: np.ndarray,
```
Comment (jmichelp) [inline scaaml/io/resume_kti.py:43]: Is there a case when we would call `create_resume_kti()` without almost immediately calling `ResumeKTI.__init__()` after?
If not I suggest to call the constructor at the end of `create_resume_kti()`.

Comment (kralka) [reply scaaml/io/resume_kti.py]: Done.

File: tests/io/test_resume_kti.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    create_resume_kti(keys=KEYS,
+                      texts=TEXTS,
+                      shard_length=SHARD_LENGTH,
+                      kt_filename=tmp_path / KT_FILENAME,
+                      progress_filename=tmp_path / PROGRESS_FILENAME)
+    # Check that the files exist now
+    assert os.path.isfile(tmp_path / KT_FILENAME)
+    assert os.path.isfile(tmp_path / PROGRESS_FILENAME)
+
+    resume_kti = ResumeKTI(kt_filename=tmp_path / KT_FILENAME,
```
Comment (jmichelp) [inline tests/io/test_resume_kti.py:25]: This line becomes obsolete now that `create_resume_kti()` directly constructs the object. Merge with line 16

File: tests/io/test_resume_kti.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+def iterate_for(tmp_path, n):
+    # Iterates for n iterations, then resumes
+    create_resume_kti(keys=KEYS,
+                      texts=TEXTS,
+                      shard_length=SHARD_LENGTH,
+                      kt_filename=tmp_path / KT_FILENAME,
+                      progress_filename=tmp_path / PROGRESS_FILENAME)
+
+    resume_kti1 = ResumeKTI(kt_filename=tmp_path / KT_FILENAME,
```
Comment (jmichelp) [inline tests/io/test_resume_kti.py:48]: Same here: can be merged with the line above.

File: tests/io/test_resume_kti.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    # Attempt to overwrite
+    inc_keys = KEYS + 1
+    inc_texts = TEXTS + 1
+    create_resume_kti(keys=inc_keys,
+                      texts=inc_texts,
+                      shard_length=SHARD_LENGTH,
+                      kt_filename=tmp_path / KT_FILENAME,
+                      progress_filename=tmp_path / PROGRESS_FILENAME)
+
+    resume_kti = ResumeKTI(kt_filename=tmp_path / KT_FILENAME,
```
Comment (jmichelp) [inline tests/io/test_resume_kti.py:100]: And same here: this line can be merged with the one above.

**Review by jmichelp — APPROVED** [review]

File: tests/io/test_resume_kti.py (hunk 2 above)
Comment (kralka) [reply tests/io/test_resume_kti.py]: Good point

**Pull request merged**